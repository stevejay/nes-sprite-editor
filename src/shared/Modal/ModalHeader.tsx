import React from "react";
import styles from "./ModalHeader.module.scss";
import { FiX } from "react-icons/fi";
import Button from "../Button";

type Props = {
  onClose?: () => void;
  children: React.ReactNode;
};

const ModalHeader = ({ onClose, children }: Props) => (
  <header className={styles.header}>
    <h2>{children}</h2>
    {onClose && (
      <Button
        aria-label="Close the dialog"
        icon={FiX}
        onClick={onClose}
        appearance="transparent"
      />
    )}
  </header>
);

export default ModalHeader;
