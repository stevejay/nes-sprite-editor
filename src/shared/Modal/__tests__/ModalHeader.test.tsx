import React from "react";
import { render, cleanup } from "react-testing-library";
import ModalHeader from "../ModalHeader";
import userEvent from "user-event";
import { noop } from "lodash";

afterEach(cleanup);

test("displays correctly", async () => {
  const { container } = render(
    <ModalHeader onClose={noop}>content</ModalHeader>
  );
  expect(container).toMatchSnapshot();
});

test("close button reacts to mouse click", async () => {
  const handleClose = jest.fn();
  const { getByLabelText } = render(
    <ModalHeader onClose={handleClose}>content</ModalHeader>
  );
  const button = getByLabelText("Close the dialog");
  expect(button).toBeTruthy();
  userEvent.click(button);
  expect(handleClose).toHaveBeenCalledTimes(1);
});
