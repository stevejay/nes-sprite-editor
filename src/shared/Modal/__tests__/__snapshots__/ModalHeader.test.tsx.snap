// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correctly 1`] = `
<div>
  <header
    class="header"
  >
    <h2>
      content
    </h2>
    <button
      aria-label="Close the dialog"
      class="button medium transparent"
      tabindex="0"
      type="button"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </button>
  </header>
</div>
`;
